<?xml version="1.0"?>
<launch>
  <arg name="map_frame" default="aruco"/>
  <arg name="marker_size" default="0.2"/>
  <arg name="map_config" default="$(find aruco_ros)/maps/map.yml"/>
  <arg name="camera_name" default="camera"/>
  <arg name="visualize" default="true"/>
  <arg name="publish_tf" default="true"/>

  <node pkg="aruco_ros" type="map_single" name="aruco_node" output="screen">
    <remap from="/camera_info" to="/$(arg camera_name)/camera_info"/>
    <remap from="/image" to="/$(arg camera_name)/image_rect_color"/>
    <param name="map_config" type="string" value="$(arg map_config)"/>
    <param name="map_frame" type="string" value="$(arg map_frame)"/>
    <param name="marker_size" type="double" value="$(arg marker_size)"/>
    <param name="draw_markers" type="bool" value="true"/>
    <param name="draw_markers_cube" type="bool" value="false"/>
    <param name="draw_markers_axis" type="bool" value="true"/>
    <param name="draw_map_axis" type="bool" value="true"/>
    <param name="image_is_rectified" value="true"/>
    <param name="camera_frame" value="$(arg camera_name)"/>
    <param name="corner_refinement" value="CORNER_SUBPIX"/>
    <param name="publish_tf" value="$(arg publish_tf)"/>
  </node>

  <node pkg="image_view" type="image_view" name="image_view" if="$(arg visualize)">
    <remap from="image" to="aruco_node/result"/>
  </node>
</launch>
